ledger.rb

```diff
@@ -1,24 +1,156 @@
+# Optional Flags
+
+# --price-db FILE | Use FILE for retrieving stored commodity prices.
+# --file FILE     | Read FILE as a ledger file.
+# --sort VEXPR    | Sort a report using VEXPR.
+
 require 'thor'
+require 'colorize'
 require_relative 'parser'
 
 class Ledger < Thor
+  class_option :file, type: :string, default: 'index.ledger'
+
+  def initialize(*args)
+    super
+    @parser = Parser.new
+    @parser.parse_ledger(options[:file])
+    @balances = {}
+  end
+
   desc "register", "The register command displays all the postings occurring in a single account, line by line."
   def register
-    puts "register command"
+    @parser.parsed_file.each do |transaction|
+      register_title_line(transaction[:date], transaction[:description])
+
+      transaction[:accounts].each do |account|
+        if @balances.include?(account[:currency])
+          @balances[account[:currency]] += account[:amount]
```

Comment:
if you have time, can extract these lines along with `register_title_line` method into a separate class so you can do something like

```
def register
  register = Register.new(@parser)
  @balances = register.call
  register_balance(@balances)
end
```

You can use whatever name you want but try to follow this structure